os/hal/ports/STM32/LLD/FDCANv1/hal_can_lld.c

```diff
@@ -134,7 +134,7 @@ static bool fdcan_clock_stop(CANDriver *canp) {
   canp->fdcan->CCCR |= FDCAN_CCCR_CSR;
   start = osalOsGetSystemTimeX();
   end = osalTimeAddX(start, TIME_MS2I(TIMEOUT_INIT_MS));
-  while ((canp->fdcan->CCCR & FDCAN_CCCR_CSA) != 0U) {
+  while ((canp->fdcan->CCCR & FDCAN_CCCR_CSA) == 0U) {

```

Comment:
This was a bug - the datasheet states that the CSA bit will be _set_ when the device is ready for clock stop.  Thus, we should spin while it's zero.